core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -35,12 +35,34 @@ class TaskInfo(
      */
     val index: Int,
     val attemptNumber: Int,
+    /**
+     * The actual RDD partition ID in this task.
+     * The ID of the RDD partition is always same across task attempts.
+     * This will be -1 for historical data, and available for all applications since 3.3.
+     */
+    val partitionId: Int,
     val launchTime: Long,
     val executorId: String,
     val host: String,
     val taskLocality: TaskLocality.TaskLocality,
     val speculative: Boolean) {
 
+  def this(
```

Comment:
can we add some comments to say this is for backward compatibility?

Reply:
Sure